forecasting/src/autogluon/forecasting/models/gluonts_model/abstract_gluonts/abstract_gluonts_model.py

```diff
@@ -0,0 +1,291 @@
+        else:
+            quantiles = [str(q) for q in quantiles]
+        result_dict = {}
+        predicted_targets = list(self.model.predict(data))
+        if isinstance(predicted_targets[0], QuantileForecast):
+            status = [0 < float(quantiles[i]) < 1 and str(quantiles[i]) in predicted_targets[0].forecast_keys for i in range(len(quantiles))]
+        elif isinstance(predicted_targets[0], SampleForecast):
+            transformed_targets = []
+            for forecast in predicted_targets:
+                tmp = []
+                for quantile in quantiles:
+                    tmp.append(forecast.quantile(quantile))
+                transformed_targets.append(QuantileForecast(forecast_arrays=np.array(tmp),
+                                                            start_date=forecast.start_date,
+                                                            freq=forecast.freq,
+                                                            forecast_keys=quantiles,
+                                                            item_id=forecast.item_id))
+            predicted_targets = copy.deepcopy(transformed_targets)
+            status = [0 < float(quantiles[i]) < 1 and str(quantiles[i]) in predicted_targets[0].forecast_keys for i in range(len(quantiles))]
+        else:
+            raise TypeError("DistributionForecast is not yet supported.")
+
+        if not all(status):
+            raise ValueError("Invalid quantile value.")
+        if isinstance(data, pd.DataFrame):
+            index = data.get_index()
+        else:
+            index = [i["item_id"] for i in data]
+        for i in range(len(index)):
+            tmp_dict = {}
+            for quantile in quantiles:
+                tmp_dict[quantile] = predicted_targets[i].quantile(str(quantile))
+            df = pd.DataFrame(tmp_dict)
+            df.index = pd.date_range(start=predicted_targets[i].start_date,
+                                     periods=self.params["prediction_length"],
+                                     freq=self.params["freq"])
+            result_dict[index[i]] = df
+        return result_dict
+
+    def predict_for_scoring(self, data, num_samples=100):
+        forecast_it, ts_it = make_evaluation_predictions(dataset=data,
+                                                         predictor=self.model,
+                                                         num_samples=num_samples)
+        return list(tqdm(forecast_it, total=len(data))), list(tqdm(ts_it, total=len(data)))
+
+    def score(self, data, metric=None, num_samples=100):
+        """
+        metric: if metric is None, we will by default use mean_wQuantileLoss for scoring.
+                should be one of "MASE", "MAPE", "sMAPE", "mean_wQuantileLoss"
+        """
+        if metric is None:
+            metric = self.eval_metric
+        if metric is not None and metric not in ["MASE", "MAPE", "sMAPE", "mean_wQuantileLoss"]:
```

Comment:
you should define a constant:
`SUPPORTED_FORECASTING_METRICS = ["MASE", "MAPE", "sMAPE", "mean_wQuantileLoss"]`

and then use this constant throughout the code instead